apps/src/netsim/DashboardUser.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2015 Code.org
+ * http://code.org/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Interface to dashboard user data API.
+ */
+
+/* jshint
+ funcscope: true,
+ newcap: true,
+ nonew: true,
+ shadow: false,
+ unused: true,
+
+ maxlen: 90,
+ maxparams: 3,
+ maxstatements: 200
+ */
+/* global $ */
+'use strict';
+
+// TODO (bbuchanan): Convert this whole file to use standard Promise/Deferred code
+
+var _ = require('../utils').getLodash();
+
+/**
+ * Represents a Dashboard user account - could be a teacher, a student, etc.
+ * @constructor
+ */
+var DashboardUser = function () {
```

Comment:
In my discussion with Geoff eariler, we talked making current user info available in a way that might render this class entirely unnecessary.  I'm not sure that'll be done for this PR though.